File: PongSharp/src/Views/GameUI.cs
using PongSharp.Data;

namespace PongSharp.Views;

internal class GameUI(Screen screen, Int2 screenSize)
{
    public void DrawProgress(GameProgress gameProgress)
    {
        var scoreYPosition = 2;
        var scoreXOffset = 3;
        var screenHalfWidth = screenSize.X / 2;

        screen.DrawNumber(new Int2(screenHalfWidth - scoreXOffset, scoreYPosition), gameProgress.LeftPlayerScore);
        screen.DrawNumber(new Int2(screenHalfWidth + scoreXOffset, scoreYPosition), gameProgress.RightPlayerScore);
    }
}

File: src/PongGame/Views/GameUI.c
#include "GameData.h"
#include "Screen.h"

void GameUI_DrawProgress()
{
    int scoreYPosition = 2;
    int scoreXOffset = 3;
    int screenHalfWidth = GameStaticData.ScreenSize.X / 2;
    Int2 leftScorePosition = {screenHalfWidth - scoreXOffset, scoreYPosition};
    Int2 rightScorePosition = {screenHalfWidth + scoreXOffset, scoreYPosition};

    Screen_DrawNumber(leftScorePosition, GameProgressData.LeftPlayerScore);
    Screen_DrawNumber(rightScorePosition, GameProgressData.RightPlayerScore);
}